azurelinuxagent/common/persist_firewall_rules.py

```diff
@@ -143,16 +143,30 @@ def __verify_firewall_rules_enabled(self):
 
         return True
 
+    def __remove_firewalld_rules(self):
+        try:
+            AddFirewallRules.remove_firewalld_rules(self._dst_ip, self._uid)
+        except Exception as error:
+            logger.verbose(
+                "failed to remove rule using firewalld.service: {0}".format(ustr(error)))
+
+    def __add_firewalld_rules(self):
+        try:
+            AddFirewallRules.add_firewalld_rules(self._dst_ip, self._uid)
+        except Exception as error:
```

Comment:
Why is exception handling needed here? This is kind of breaking the flow of the code. For example, if we're unable to add the rules, you log the error below and handle the exception. Then as soon as you exit this function, this subsequent statement will be logged -  
> logger.info("Successfully added the firewall commands using firewalld.service")

Which is completely wrong since we were infact unable to add the firewall rules. 

Maybe I'm missing something, but could you please explain a bit on why the error handling is needed here?

Reply:
I didn't look at that logger.info. That's good point we should return if we unable to add rules. 

`Maybe I'm missing something, but could you please explain a bit on why the error handling is needed here? `

As per my understanding, these are shell commands, you would expect exit codes between 1-255. In this case you don't see exit code 1 but very likely you can expect others when shell command didn't run properly. Correct me if I'm wrong?

Reply:
yes that's correct. But this code was kinda designed that way. Like its supposed to throw if we fail to execute those commands as that defines how the whole sequence of operations work. Does that make sense? If not we can get over it on a call if you want.

Reply:
I understood the whole sequence and it shouldn't go to next operation and log successful message when exception happens. I was explaining why I decided to do exception handling. I will handle that sequence by return when exception happens.

Reply:
I just checked the code we have upstream error handling so no need to do it here, thanks